subscriptions-matcher/src/main/java/io/arlas/subscriptions/service/ProductService.java

```diff
@@ -68,8 +68,11 @@ void processMatchingProducts(SubscriptionEvent event, List<Hit> subscriptions) t
                     userSubscription.setId((String) ((Map<String, Object>) hit.getData()).get("id"));
                     logger.trace("indexedUserSubscription: " + userSubscription.toString());
                     String f = searchFilter
-                            + "&" + userSubscription.subscription.hits.filter
-                            + "&" + userSubscription.subscription.hits.projection;
+                            + "&" + userSubscription.subscription.hits.filter;
+                    if (userSubscription.subscription.hits.projection != null
```

Comment:
You may use a StringUtils, if available in the project, to avoid the double check (like `StringUtils.isNotBlank(...)`)

Reply:
Fixed